docs/querying/nested-columns.md

```diff
@@ -0,0 +1,503 @@
+```
+
+### Using other transform functions
+
+Druid provides the following additional transform functions:
+
+- `PARSE_JSON`: Deserializes a string value into a JSON object.
+- `TO_JSON_STRING`: Performs the operation of `TO_JSON` and then serializes the value into a string.
+
+#### Example query: Parse and deserialize data
+
+ The following query illustrates how to use the transform functions to parse and deserialize data.
+
+![Parse and deserialize data](../assets/nested-parse-deserialize.png)
+
+```sql
+SELECT
+  PARSE_JSON('{"x":"y"}'),
+  TO_JSON_STRING('{"x":"y"}'),
+  TO_JSON_STRING(PARSE_JSON('{"x":"y"}'))
+```
+
+Example query results:
+
+```json
+[["EXPR$0","EXPR$2","EXPR$3"],["COMPLEX<json>","STRING","STRING"],["OTHER","VARCHAR","VARCHAR"],["{\"x\":\"y\"}","\"{\\\"x\\\":\\\"y\\\"}\"","{\"x\":\"y\"}"]]
+```
+
+### Using helper operators
+
+The `JSON_KEYS` and `JSON_PATHS` functions are helper operators that you can use to examine JSON object schema. Use them to plan your queries, for example to work out which paths to use in `JSON_VALUE`.
+
+#### Example query: Examine JSON object schema
+
+ The following query illustrates how to use the helper operators to examine a nested data object.
+
+![Examine JSON object schema](../assets/nested-examine-schema.png)
+
+```sql
+SELECT
+  ARRAY_CONCAT_AGG(DISTINCT JSON_KEYS(shipTo, '$.')),
+  ARRAY_CONCAT_AGG(DISTINCT JSON_KEYS(shipTo, '$.address')),
+  ARRAY_CONCAT_AGG(DISTINCT JSON_PATHS(shipTo))
+FROM nested_data_example
+```
+
+Example query results:
+
+```json
+[["EXPR$0","EXPR$1","EXPR$2","EXPR$3"],["COMPLEX<json>","COMPLEX<json>","STRING","STRING"],["OTHER","OTHER","VARCHAR","VARCHAR"],["{\"x\":\"y\"}","\"{\\\"x\\\":\\\"y\\\"}\"","\"{\\\"x\\\":\\\"y\\\"}\"","{\"x\":\"y\"}"]]
+```
+
+## Known issues
+
+Before you start using the nested columns feature, consider the following known issues:
+
+- Directly using `COMPLEX<json>` typed columns and expressions is not well integrated into the Druid query engine. It can result in errors or undefined behavior when grouping and filtering, and when you use using `COMPLEX<json>` objects as inputs to aggregators. As a workaround, consider using `TO_JSON_STRING` to coerce the values to strings before you perform these operations.
```

Comment:
> `COMPLEX<json>` typed columns

should probably have a hyphen before typed like "array-typed" in the line below.

Or to avoid weird formatting you could say something like 
> `COMPLEX<json>` columns

Reply:
Done